pyqtorch/modules/primitive.py

```diff
@@ -427,3 +434,30 @@ def __init__(self, qubits: ArrayLike, n_qubits: int):
         print(result)
         """
         super().__init__("SWAP", qubits, n_qubits)
+
+
+class Toffoli(ControlledOperationGate):
+    def __init__(self, qubits: ArrayLike, n_qubits: int):
+        """
+        Represents a multi qubit controlled toffoli gate in a quantum circuit.
+        This gate performs a NOT operation only if all the control qubits are in state 1.
+        Arguments:
+            qubits (ArrayLike): The first n-1 qubits will be considered as the control
+            qubits and the last one will be the target qubit of the Toffoli gate.
+
+        Examples:
+        ```python exec="on" source="above" result="json"
+        import torch
+        import pyqtorch.modules as pyq
+
+        # Create a Toffoli gate with 3 control qubits.
+        toffoli_gate = pyq.Toffoli(qubits=[0, 1, 2, 3], n_qubits=4)
+
+        # Create a zero state
+        toffoli_state = pyq.zero_state(n_qubits=4)
+
+        # Apply the CSWAP gate to the zero state
+        result = toffoli_gate(toffoli_state)
+        print(result)
```

Comment:
Typo on the CSWAP and I would change this example to initializing a one state just so the toffoli actually does something.

Reply:
Yep you're right !